Neos.ContentRepository.Core/Classes/Projection/ContentGraph/Node.php

```diff
@@ -75,9 +75,23 @@ public function __construct(
          * @return PropertyCollectionInterface Property values, indexed by their name
          */
         public PropertyCollectionInterface $properties,
+        /**
+         * The node's name. The name is guaranteed to be for tethered nodes.
+         * For the regular classification, it can be set optionally.
+         */
         public ?NodeName $nodeName,
         public Timestamps $timestamps,
     ) {
+        assert(
+            $this->classification !== NodeAggregateClassification::CLASSIFICATION_TETHERED
+            || $this->nodeName instanceof NodeName,
+            'The NodeName must be set if the Node is tethered.'
+        );
+        assert(
+            !$this->nodeType instanceof NodeType
+            || $this->nodeType->name->equals($this->nodeTypeName),
+            'The name of the NodeType must equal the NodeTypeName of the Node.'
```

Comment:
in case you leave this in:
```suggestion
            'The name of the NodeType must be equal to the NodeTypeName of the Node.'
```